File: Leetcode 0495 Teemo Attacking.py
# 20250807
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries:
            return 0
        elif len(timeSeries) == 1:
            return duration
        # we add last duration in here
        ans = duration
        for i in range(len(timeSeries)-1):
            if timeSeries[i+1] - timeSeries[i] < duration:
                ans += timeSeries[i+1] - timeSeries[i]
            else:
                ans += duration

        return ans

File: Leetcode 0495 Teemo Attacking.go
// 20250807
func findPoisonedDuration(timeSeries []int, duration int) int {
    if len(timeSeries) == 0{
        return 0
    } else if len(timeSeries)==1{
        return duration
    }
    // last attack
    ans := duration
    for i:=0; i<len(timeSeries)-1; i++{
        if timeSeries[i+1] - timeSeries[i] < duration{
            ans += (timeSeries[i+1] - timeSeries[i])
        } else{
            ans += duration
        }
    }
    return ans
}
